### PR TITLE
refactors master ctrl mgr main function

### PR DESCRIPTION
**Pull request by p0lyn0mial**
**What this PR does / why we need it**: simply refactors master ctrl mgr main function

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
See #2929 

**Release note**:

```
NONE
```

### CODE REVIEW

File: api/cmd/master-controller-manager/controllers.go [hunk 1]
```diff
@@ -0,0 +1,167 @@
...
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/tools/clientcmd"
+	"sigs.k8s.io/controller-runtime/pkg/manager"
+)
+
+type controllerCreator func(*controllerContext) (runnerFn, error)
+type runnerFn func(workerCount int, stopCh <-chan struct{}) error
+
+func noop(workerCount int, stopCh <-chan struct{}) error { <-stopCh; return nil }
```
Comment (zreigz) [inline api/cmd/master-controller-manager/controllers.go:26]: worker count is not used

File: api/cmd/master-controller-manager/main.go [hunk 2]
```diff
@@ -163,48 +121,13 @@ func main() {
...
-			}
-
-			if err := userprojectbinding.Add(mgr); err != nil {
-				return err
-			}
-			if err := serviceaccount.Add(mgr); err != nil {
-				return err
-			}
-
-			if err := mgr.Start(ctrlCtx.stopCh); err != nil {
```
Comment (zreigz) [inline api/cmd/master-controller-manager/main.go:201]: controller manager is not started in the new code
Metrics are not exposed: `http://localhost:8085/metrics`

**Review by zreigz — CHANGES_REQUESTED** [review]

File: api/cmd/master-controller-manager/main.go (hunk 2 above)
Comment (zreigz) [reply api/cmd/master-controller-manager/main.go:201]: my mistake, works fine

File: api/cmd/master-controller-manager/controllers.go (hunk 1 above)
Comment (p0lyn0mial) [reply api/cmd/master-controller-manager/controllers.go:26]: `noop` implements `runnerFn`, the signature must match.

**Pull request merged**